src/clientfactory.ts

```diff
@@ -91,4 +91,12 @@ export class DiscordClientFactory {
       return this.botClient;
     }
   }
+
+  public async UserIsPuppeted(userId: string = null): Promise<boolean> {
+      if (this.clients.has(userId)) {
+          return true;
+      }
+      const discordIds = await this.store.get_user_discord_ids(userId);
+      return discordIds.length !== 0;
```

Comment:
Add in-memory caching so that we don't have to hit the store each time?

Reply:
We should be sufficiently speedy not to have to worry about this for now. I'll add caching if it becomes a problem.